File: ordermanager-backend/src/main/java/com/pr/ordermanager/security/repository/RoleRepository.java
package com.pr.ordermanager.security.repository;

import com.pr.ordermanager.security.entity.GrantedRole;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RoleRepository extends JpaRepository<GrantedRole, Long> {
    GrantedRole findByAuthority(String authority);
}

File: ordermanager-backend/src-node/src/security/repositories/role.repository.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { GrantedRoleEntity } from '../entities/granted-role.entity';

@Injectable()
export class RoleRepository {
  constructor(@InjectRepository(GrantedRoleEntity) private readonly repo: Repository<GrantedRoleEntity>) {}

  findByAuthority(authority: string): Promise<GrantedRoleEntity | null> {
    return this.repo.findOne({ where: { authority } });
  }

  save(role: GrantedRoleEntity): Promise<GrantedRoleEntity> {
    return this.repo.save(role);
  }
}
